src/vs/platform/native/electron-main/nativeHostMainService.ts

```diff
@@ -491,12 +493,52 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 
 	async openExternal(windowId: number | undefined, url: string): Promise<boolean> {
 		this.environmentMainService.unsetSnapExportedVariables();
-		shell.openExternal(url);
-		this.environmentMainService.restoreSnapExportedVariables();
+		try {
+			if (matchesSomeScheme(url, Schemas.http, Schemas.https)) {
+				this.openExternalBrowser(url);
+			} else {
+				shell.openExternal(url);
+			}
+		} finally {
+			this.environmentMainService.restoreSnapExportedVariables();
+		}
```

Comment:
I'm wondering, should this be using `await` before restoring the snap exported variables? 

```suggestion
		try {
			if (matchesSomeScheme(url, Schemas.http, Schemas.https)) {
				await this.openExternalBrowser(url);
			} else {
				await shell.openExternal(url);
			}
		} finally {
			this.environmentMainService.restoreSnapExportedVariables();
		}
```

Reply:
Unless that fixes a bug, I would not change this just by speculating.